src/connectors/numeric-menu/connectNumericMenu.ts

```diff
@@ -4,146 +4,157 @@ import {
+}
+
+export type NumericMenuRenderer<TNumericMenuWidgetParams> = Renderer<
+  NumericMenuRendererOptions<
+    NumericMenuConnectorParams & TNumericMenuWidgetParams
+  >
+>;
+
+export type NumericMenuWidgetFactory<TNumericMenuWidgetParams> = WidgetFactory<
+  NumericMenuConnectorParams & TNumericMenuWidgetParams
+>;
+
+type NumericMenuConnector = <TNumericMenuWidgetParams>(
+  render: NumericMenuRenderer<TNumericMenuWidgetParams>,
+  unmount?: () => void
+) => NumericMenuWidgetFactory<TNumericMenuWidgetParams>;
+
+const connectNumericMenu: NumericMenuConnector = (
+  renderFn,
+  unmountFn = noop
+) => {
   checkRendering(renderFn, withUsage());
 
-  return (widgetParams = {}) => {
-    const { attribute, items, transformItems = x => x } = widgetParams;
+  return widgetParams => {
+    const {
+      attribute = '',
+      items = [],
+      transformItems = (x => x) as ParamTransformItems,
+    } = widgetParams || {};
 
-    if (!attribute) {
+    if (attribute === '') {
       throw new Error(withUsage('The `attribute` option is required.'));
     }
 
-    if (!items) {
+    if (!items || items.length === 0) {
       throw new Error(
         withUsage('The `items` option expects an array of objects.')
       );
     }
 
+    type ConnectorState = {
+      refine?: Refine;
+      createURL?: (state: SearchParameters) => (facetValue: string) => string;
+      prepareItems?: (state: SearchParameters) => Item[];
+    };
+
+    const connectorState: ConnectorState = {};
+
     return {
       $$type: 'ais.numericMenu',
 
       init({ helper, createURL, instantSearchInstance }) {
-        this._refine = facetValue => {
-          const refinedState = refine(
-            helper.state,
-            attribute,
-            items,
-            facetValue
-          );
+        connectorState.refine = facetValue => {
+          const refinedState = refine(helper.state, attribute, facetValue);
           helper.setState(refinedState).search();
         };
 
-        this._createURL = state => facetValue =>
-          createURL(refine(state, attribute, items, facetValue));
+        connectorState.createURL = state => facetValue =>
+          createURL(refine(state, attribute, facetValue));
 
-        this._prepareItems = state =>
+        connectorState.prepareItems = state =>
           items.map(({ start, end, label }) => ({
             label,
-            value: window.encodeURI(JSON.stringify({ start, end })),
-            isRefined: isRefined(state, attribute, { start, end }),
+            value: (window as any).encodeURI(JSON.stringify({ start, end })),
+            isRefined: isRefined(state, attribute, { start, end, label }),
           }));
 
         renderFn(
           {
-            createURL: this._createURL(helper.state),
-            items: transformItems(this._prepareItems(helper.state)),
+            createURL: connectorState.createURL(helper.state),
+            items: transformItems(connectorState.prepareItems!(helper.state)),
```

Comment:
`!` shouldn't be required here.

```suggestion
            items: transformItems(connectorState.prepareItems(helper.state)),
```